### PR TITLE
CAMEL-19507 - Upgrade Wildfly Elytron to version 1.20.4.Final

### PR DESCRIPTION
**Pull request by oscerd**
# Description

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [ ] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

File: catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/schemas/camel-spring.xsd
Comment (essobedo) [inline catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/schemas/camel-spring.xsd:1]: Doesn't seem to be related, right?

**Review by essobedo — APPROVED** [review]

File: catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/schemas/camel-spring.xsd
Comment (oscerd) [reply catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/schemas/camel-spring.xsd:1]: I think we merged something wrong in a regen action and now it's back to the correct version.

**Review by davsclaus — APPROVED** [review]
